[PKM-10] Presentation Layer Setup

yohancai opened the pull request:
- initial view of Pokemon List
- 
![image](https://user-images.githubusercontent.com/58543326/182747614-289e62fd-63ac-48a7-a85e-be3951b3f7e6.png)

Commit:
[PKM-10] Presentation Layer Setup

- initial view of Pokemon List

JotamSleonardo commented:
Please send screenshots or video

JotamSleonardo commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -7,5 +7,72 @@ class PokemonOverviewPage extends StatelessWidget {
+                  borderRadius: BorderRadius.circular(20),
+                  color: Colors.white,
+                ),
+                child: Stack(
```
Create a separate widget for this pokemon cell.

JotamSleonardo commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -7,5 +7,72 @@ class PokemonOverviewPage extends StatelessWidget {
+                ),
+                child: Stack(
+                  children: const [
+                    Positioned(
```
Much better to create rows and column. This can have overlapping

JotamSleonardo commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -7,5 +7,72 @@ class PokemonOverviewPage extends StatelessWidget {
+          crossAxisCount: 2,
+          children: List.generate(20, (index) {
+            return GestureDetector(
+              child: Container(
```
You can add the onTap parameter for this but you need to add //TODO for the action

JotamSleonardo reviewed (changes requested):
Please see PR feedbacks

Commit:
[PKM-10] Presentation Layer Setup

- created ontap
- created initial Pokemon tile widget
- change format into rows and columns

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_tile.dart:
```diff
@@ -0,0 +1,65 @@
+import 'package:go_router/go_router.dart';
+import 'package:pokedex/utils/strings.dart' as str;
+
+class PokemonTile extends StatefulWidget {
```
- do you really need to make this stateful? 🤔 
- kindly rename this to PokemonTileItem 🙂 
- is this something to be reused for favorites as well right? If so, kindly move this to your general widget directory 🙂

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_tile.dart:
```diff
@@ -0,0 +1,65 @@
+          ],
+        ),
+      ),
+      onTap: () => context.pop(),
```
I would suggest to place this `onTap` at the top of `child` param so it will easily be accessed at first glance 🙂

shawn-ffufm commented on pubspec.lock:
```diff
@@ -511,6 +511,13 @@ packages:
       url: "https://pub.dartlang.org"
     source: hosted
     version: "1.8.2"
+  stack:
```
for most of the cases I think it's not recommended for us to push any changes from `pubspec.lock` since this is generated after we performed `flutter pub get` based on your local machine's current config, unless @raelsj will require to for any important reason I would say 🤔

yohancai replied:
should I add this to .gitignore?

shawn-ffufm commented on pubspec.yaml:
```diff
@@ -29,6 +29,9 @@ dependencies:
   path_provider: ^2.0.11
 
+  #UI
+  stack: ^0.0.1
```
any insight what's this for? 🤔

yohancai replied:
this was the one I used initially for the UI but boss @Jotam suggest to use rows and columns as this is prone to overlaps

shawn-ffufm replied:
so will it still be used anyways? If not feel free to remove 🙂

yohancai replied:
yes it will not be used anymore

shawn-ffufm reviewed (changes requested):
@yohancai see minor pr feedback 🙂

Commit:
[PKM-10] Presentation Layer Setup

- changed pokemon tile item into stateless
- refactor pokemon tile item into reusable widget

shawn-ffufm commented on lib/widgets/pokemon_tile.dart:
```diff
@@ -0,0 +1,62 @@
+import 'package:go_router/go_router.dart';
+import 'package:pokedex/utils/strings.dart' as str;
+
+class PokemonTileItem extends StatelessWidget {
```
@yohancai you forgot to rename your dart file as well 😅

shawn-ffufm reviewed (changes requested):
@yohancai see minor feedback

shawn-ffufm commented on lib/widgets/pokemon_tile.dart:
```diff
@@ -0,0 +1,62 @@
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: () => context.pop(),
```
where's the included todo comment for this now? 🤔

Commit:
[PKM-10] Presentation Layer Setup

- refactor pokemon tile item

Commit:
[PKM-10] Presentation Layer Setup

- removed trailing comma

shawn-ffufm reviewed (approved):
![image](https://user-images.githubusercontent.com/83454273/182769733-f382e299-a375-4ad9-ac91-c4cf4cc1fab0.png)

raelsj commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -1,11 +1,28 @@
+          gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(
+            maxCrossAxisExtent: 190,
+            childAspectRatio: 10 / 9,
+            crossAxisSpacing: 10,
```
please create a private constant above the file for this number 10.

raelsj commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -1,11 +1,28 @@
+          padding: const EdgeInsets.all(10),
+          gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(
+            maxCrossAxisExtent: 190,
+            childAspectRatio: 10 / 9,
```
can u explain this

yohancai replied:
it creates the tile template for the pokemon

Commit:
[PKM-10] Presentation Layer Setup

- created private variable for ten

raelsj commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -1,11 +1,30 @@
-  });
+  const PokemonOverviewPage({super.key});
+
+  static const double _ten = 10;
```
hahaha. no, name it properly. what is 10?

raelsj commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -1,11 +1,30 @@
+        body: GridView.builder(
+          padding: const EdgeInsets.all(10),
+          gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(
+            maxCrossAxisExtent: 190,
```
do the same here.

Commit:
[PKM-10] Presentation Layer Setup

- created static private variables
